packages/front-end/services/UserContext.tsx

```diff
@@ -262,6 +262,15 @@ export function UserContextProvider({ children }: { children: ReactNode }) {
     }
   }, [data?.email, data?.userId]);
 
+  // Validate organization id with license
+  useEffect(() => {
+    if (currentOrg?.organization?.id && data?.license?.org) {
+      if (currentOrg.organization.id !== data.license.org) {
+        throw new Error("Organization id does not match license");
+      }
+    }
+  }, [currentOrg?.organization, data?.license]);
```

Comment:
Move this to the back-end.  Add a new middleware that compares the org id in the request and the org id in the license key. If they both exist and are different, return an API error.

Reply:
@jdorn  I added a new backend middleware for this for pages requiring auth. It relies on the x-organization-id header.

There didn't appear to be an out-of-the-box way to control the status code (410), as express middlewares are designed to fail/cancel by passing an Error to `next(e)`.

The user experience is pretty jagged on the UI side when this fails. Random error messages written onto the page at random positions (or sometimes not at all). Sometimes the error is rendered out, sometimes it's caught and a generic error is shown. Not sure we want to optimize given who will typically be seeing this error.